### PR TITLE
Convert PHPUnit annotations to attributes

### PR DESCRIPTION
**Pull request by chr-hertel**
Rector did the work and I cleaned up. Not sure if you're open for this kind of PRs, but was my sunday meditation anyways :)

### CODE REVIEW

**Comment by greg0ire** [comment]
By all means, we're open to such PRs. Can you take a look at phpcs?

**Comment by derrabus** [comment]
Looks good so far. One thing: Doesn't PHPUnit 10 have an attribute equivalent for `@requires` or did Rector just not catch those?

> Can you take a look at phpcs?

PHPCS fails because a test has been commented out and because the following test does not have a PHPDoc block anymore, it assumes that the commented-put test is a badly formatted PHPDoc block.

Here's my take at fixing the issue on 2.14.x by skipping the test instead of commenting it out: #10560. Alternatively, we could just delete that test.

**Comment by chr-hertel** [comment]
> Doesn't PHPUnit 10 have an attribute equivalent for @requires or did Rector just not catch those?

It does, Rector missed it or wasn't able. I will have another look 👍

**Comment by derrabus** [comment]
The PHPCS failure should be gone after a rebase.

**Comment by greg0ire** [comment]
> The PHPCS failure should be gone after a rebase.

You mean one failure out of three I think. @chr-hertel please do check it works before force-pushing.

File: tests/Doctrine/Tests/ORM/Functional/SQLFilterTest.php [hunk 1]
```diff
@@ -25,8 +25,10 @@
 use Doctrine\Tests\Models\Company\CompanyManager;
 use Doctrine\Tests\Models\Company\CompanyOrganization;
 use Doctrine\Tests\Models\Company\CompanyPerson;
+use Doctrine\Tests\ORM\Functional\Ticket\MyEntity;
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/SQLFilterTest.php]: This needs to be reverted

**Comment by greg0ire** [comment]
This should help: https://github.com/doctrine/orm/pull/10563

**Comment by derrabus** [comment]
Ah, I missed the other test. Thanks, @greg0ire.

**Comment by derrabus** [comment]
Browsing through all of the `RequiresPhp` attributes, I realized that they could actually all be removed because we don't run the test suite on PHP < 8.1 on this branch. I did remove a lot of those `@requires` annotations previously, but merge-ups keep reintroducing them. 😓 

But we can remove those in a follow-up, no need to block a merge because of that.

Great work. 👍🏻

**Review by greg0ire — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Pull request merged**